### PR TITLE
Removed bootstrap, functionality improvements

### PR DESCRIPTION
**Pull request by BraffZackland**
Quite a lot of additions and changes over this past week.

Model changes, added improvements to template function and data validation.

Began work on custom field export and data validation.

### CODE REVIEW

File: src/google_scripts/export.gs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+      //pass values to mapper function
+      //mapper iterates and assigns the values number based on the list order
+      if(i === 4.0){ cell = mapper(cell, reqs.dropdowns['Type']) }
+
+      if(i === 5.0){ xObj['ImportanceId'] = mapper(cell, reqs.dropdowns['Importance']) }
+
+      if(i === 6.0){ xObj['StatusId'] = mapper(cell, reqs.dropdowns['Status']) }
+
+      if (i === 8.0){ xObj['AuthorId'] = mapper(cell, users) }
```
Comment (BraffZackland) [inline src/google_scripts/export.gs]: Field 8 and 9 reference two different object properties, I wouldn't be able to combine them in this way.

File: src/google_scripts/export.gs [hunk 2]
```diff
@@ -0,0 +1,220 @@
+//export function pulled from Code.gs
+//takes item {cell}, list {array}, and isObj {bool}
+//isObj is true if list is an object, i.e in the case of the users array
+
+function exporter(data){
+  var ss = SpreadsheetApp.getActiveSpreadsheet()
```
Comment (BraffZackland) [inline src/google_scripts/export.gs]: Changed all occurrences of ss to spreadSheet

File: src/google_scripts/export.gs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  var sheet = ss.getSheets()[0];
+  //number of cells in a row
+  var range = sheet.getRange(data.templateData.requirements.cellRange)
```
Comment (BraffZackland) [inline src/google_scripts/export.gs]: Yeah this is all very inconsistent , I'll go through line by line and add semicolons where they're missing!

File: src/google_scripts/export.gs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+function mapper(item, list){
+  var val = 1;
+  for (var i = 0; i < list.length; i++){
+    if (item == list[i][1]) {val = list[i][0]}
+  }
+  return val;
+}
+
+//gets full model data and custom properites cell range
+function customBuilder(data, rowRange){
```
Comment (BraffZackland) [inline src/google_scripts/export.gs]: changed to cutomHeaderRowBuilder

File: src/google_scripts/import.gs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    range.getCell(ss_i, 5).setValue(data[i].RequirementTypeName);
+    range.getCell(ss_i, 6).setValue(data[i].ImportanceName);
+    range.getCell(ss_i, 7).setValue(data[i].StatusName);
+    range.getCell(ss_i, 8).setValue(data[i].EstimatePoints);
+    range.getCell(ss_i, 9).setValue(data[i].AuthorName);
+    range.getCell(ss_i, 10).setValue(data[i].OwnerName);
+    range.getCell(ss_i, 11).setValue(data[i].ComponentId);
+
+    //moves the range down one row
+    range = range.offset(1, 0, 43);
```
Comment (BraffZackland) [inline src/google_scripts/import.gs]: refactored

File: src/google_scripts/main.gs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+      .setTitle('Inflectra Corporation');
+  SpreadsheetApp.getUi().showSidebar(ui);
+}
+
+function include(filename) {
+  return HtmlService.createHtmlOutputFromFile(filename).getContent();
+}
+
+function getProjects(currentUser){
+  var params = '/services/v5_0/RestService.svc/projects?username='
```
Comment (BraffZackland) [inline src/google_scripts/main.gs]: changed to fetcherURL

File: src/google_scripts/main.gs [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+//save function
+function save(){
+  //pop up telling the user that their data will be saved
+  var ui = SpreadsheetApp.getUi();
+  var response = ui.alert('This will save the current sheet in a new tab. Continue?', ui.ButtonSet.YES_NO);
+
+  //returns with user choice
+  if (response == ui.Button.YES) {
+    var ss = SpreadsheetApp.getActiveSpreadsheet();
```
Comment (BraffZackland) [inline src/google_scripts/main.gs]: changed to spreadSheet

File: src/google_scripts/template.gs [hunk 8]
```diff
@@ -0,0 +1,180 @@
+function templateLoader(data){
+  clearAll();
+
+  var ss = SpreadsheetApp.getActiveSpreadsheet();
```
Comment (BraffZackland) [inline src/google_scripts/template.gs]: changed to spreadSheet

File: src/google_scripts/template.gs [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+  var sheet = ss.getSheets()[0];
+  var dropdownColumnAssignments = [
+        ['Type', 'e'],['Importance', 'f'], ['Status', 'g'], ['Author', 'i'], ['Owner', 'j']
```
Comment (BraffZackland) [inline src/google_scripts/template.gs]: refactored and data model updated!

File: src/google_scripts/template.gs [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+  //set customfield cells as grey if inactive
+  var customCellRange = sheet.getRange('N3:AQ200');
+  customCellRange.setBackground('#a6a6a6');
+
+  //set column A to present a warning if the user trys to write in a value
+  var protection = reqIdRange.protect().setDescription('Exported items must not have a requirement number');
+  //set warning. Remove this to make the column un-writable
+  protection.setWarningOnly(true);
+
+  sheet.getRange('A1:M1').merge().setValue("Requirements Standard Fields").setHorizontalAlignment("center");
```
Comment (BraffZackland) [inline src/google_scripts/template.gs]: refactored and data model updated!

File: src/google_scripts/template.gs [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+  sheet.getRange('N1:AQ1').merge().setValue("Custom Fields").setHorizontalAlignment("center");
```
Comment (BraffZackland) [inline src/google_scripts/template.gs]: refactored and data model updated!

File: src/google_scripts/export.gs (hunk 1 above)
Comment (yipmarvel) [inline src/google_scripts/export.gs]: oops good point

**Pull request merged**